core/src/util/featureSelection.cpp

```diff
@@ -29,12 +32,11 @@ uint32_t FeatureSelection::colorIdentifier(const Feature& _feature, const TileID
 
 bool FeatureSelection::clearFeaturesForTile(TileID _tileID) {
 
+    std::lock_guard<std::mutex> guard(m_mutex);
```

Comment:
Do we really need to guard the whole block? `find` and `end` on `m_tileFeatures` could use const (`const auto it` can force to use a `const_iterator` on `find`).